src/main/java/org/moltimate/moltimatebackend/Structure/StructureUtils.java

```diff
@@ -0,0 +1,184 @@
+                    return group;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Estimate location of residue by averaging the locations of each
+     * atom in the residue
+     *
+     * @param residue: residue to find location of
+     * @return x, y, z coordinates of the residue
+     */
+    public static double[] getResidueLocation(Group residue) {
+        List<Atom> atoms = residue.getAtoms();
+        double[] location = new double[3];
+        for (Atom atom : atoms) {
+            double[] coords = atom.getCoords();
+            location[0] += coords[0];
+            location[1] += coords[1];
+            location[2] += coords[2];
+        }
+
+        location[0] = location[0] / atoms.size();
+        location[1] = location[1] / atoms.size();
+        location[2] = location[2] / atoms.size();
+
+        return location;
+    }
+
+    /**
+     * Find all amino acids of a certain type (ex: asp or his) inside a structure
+     *
+     * @param structure:   structure to search for amino acid in
+     * @param residueName: name of residue to search for
+     * @return list of residues with given name inside structure
+     */
+    public static List<Group> getResiduesByType(Structure structure, String residueName) {
+        ArrayList<Group> results = new ArrayList<>();
+        for (Chain chain : structure.getChains()) {
+            for (Group residue : chain.getAtomGroups(GroupType.AMINOACID)) {
+                if (residue.getChemComp().getThree_letter_code().equals(residueName)) {
+                    results.add(residue);
+                }
+            }
+        }
+        return results;
+    }
+
+    /**
+     * Get all atoms of a certain type inside a structure
+     *
+     * @param structure: structure to search for atoms in
+     * @param atomType:  type of atom to search for in structure
+     * @return a list of atoms in the structure matching the specified type
+     */
+    public static List<Atom> getAtomByType(Structure structure, String atomType) {
+        ArrayList<Atom> atoms = new ArrayList<>();
+        structure.getChains().forEach(chain ->
+                chain.getAtomGroups(GroupType.AMINOACID).forEach(group ->
+                        atoms.addAll(group.getAtoms().stream()
+                                .filter(atom -> atom.getName().equals(atomType))
+                                .collect(Collectors.toList()))));
+        return atoms;
+    }
+
+    /**
+     * Get all atoms of a certain type inside a residue
+     *
+     * @param residue:  residue to search for atoms in
+     * @param atomType: type of atom to search for in structure
+     * @return a list of atoms in the structure matching the specified type
+     */
+    public static List<Atom> getAtomByType(Group residue, String atomType) {
+
+        return residue.getAtoms().stream()
```

Comment:
Newline each same-level method when the method chain is >2 calls
```
        return residue.getAtoms()
                      .stream()
                      .filter(atom -> atom.getName().equals(atomType))
                      .collect(Collectors.toList());
```

Reply:
Done